packages/server/src/models/user-shelf.entity.ts

```diff
@@ -0,0 +1,41 @@
+import { Column, Entity, PrimaryColumn } from 'typeorm';
+import { UserShelfRoles, UserShelfActions } from '../constants/user-shelf';
+
+abstract class ShelfDuties {
```

Comment:
I'd suggest to move this to a separate table where we'll have 
role and its actions

UserShelfDuties  and UserShelfInvitation will have only role field

Reply:
Well, it would be more normalized, but would complicate purging the outdated invitations, for examle, seems cleaner to keep the invitation info in a single table.
Again, open to discussion.

Reply:
OK, I see what you're saying, and I see where the confusion comes from, let me try to explain. The `roles` and `actions` are not directly related to each other, in fact they are not related at all.

Roles are:
 - `admin` - can add other users to help with the shelf
 - `keeper` - only helps with the shelf, does not manage other users

Actions are about what a person actually does and gets notifications on:
 - `watering`
 - `hydration`
 - `fertilizing`

Any user can be asked to perform one or more `actions` on a shelf, for example:
 - Admin#1 would himself handle the watering, and so he would only get notifications on watering
 - Keeper#1 would be configured to be responsible for hydration, so would only get notificatins on hydration
 - Kepper#2 would be responsible for watering and fertilizing
 - Kepper#3 would be responsible for all the actions, so will get all the notifications
 - Admin #2 would be responsible for all the actions too, all notifications, etc...

Reply:
In this case, why do we need roles at all? we could just grant some actions and for admin grand external action "administration"

Reply:
We could, but we shouldn't, because they are logically unrelated. `watering/hydration/fertilizing` are actions a user would perform on flowers on the particular shelf. Lately notifications logic would  be based on these. At least I thought having the ability to spread the responsibilitiesis a nice feature. I also think we discussed this, but maybe it is an overkill and we should leave everyone responsible for everything.

While `administration` is something to do with managing access to the shelf, and granting the administator role to other users. Having an enum of just two roles for this might be an overkill too, because they are actually just `admin/not-admin` and might as well have an `isAdmin` binary flag in the `user_shelf` table (we wouldn't need to call it `duties` any more).

Reply:
Reworked